### ISSUE TITLE
Kudos : I can send a kudos to a user who isn't a member of chosen space

### ISSUE THREAD

**Issue by Jihed525** [Meeds-io/meeds#2610]
**Steps to reproduce the behavior:**
1. USerX is member of spaceX (hidden, closed)
2. UserY is a user (guest or internal, no matter) which isn't a member of spaceX
3. UserX wants to send a kudos to UserY:
   - from the main stream composer, by adding him in the recipient field
   - from the profile option (profile page, user popover)
4. UserX choose a space UserY is not member of

_Error:_ 
- UserY cannot access the kudos message as not member of the space

_Precision:_ 
- Ok when choosing first the space audience
- Ok when sending a kudos from a space stream
> only members of the selected space are suggested

**Expected**
Check if both member of space
   - When searching for a space, then only spaces both users are member of can be suggested
   - If then, I remove the user and I look for another user, then only users from the expected space are still suggested **(already the case and OK)**

**Comment by Jihed525** [comment]
do you confirm @srenault-meeds ?

**Comment by boubaker** [comment]
@Jihed525 do you mean, the `UserX` when attempting to send a kudos inside a space, the suggester displays other users who **aren't member of the space** ?

**Comment by Jihed525** [comment]
exactly @boubaker

**Comment by boubaker** [comment]
> exactly @boubaker

ok, may be to change the use case to specify that there is an inconsistency which allows to display users who aren't member of the space. The expected functional behavio is to list members only in fact.
BTW, can you please verify that it's reproduced in Whitepaper/MeedsV2 ACCs (wasn't able to reproduce it in fact) ?

**Comment by srenault-meeds** [comment]
I only reproduce it when sending a kudos from the main composer (general feed)
- I choose the recipient userX
- Then I choose to post the message in a space userX is not member of

However, when doing following ways, it is not a problem:
- from main composer, I choose the audience where userX is not member of, I cannot find userX in the recipient suggestor
- from a space composer, I cannot find userX in the recipient suggester

Is that right @Jihed525?

**Comment by Jihed525** [comment]
sorry i missed a key step : send kudos from user profile
![image](https://github.com/user-attachments/assets/8c913ebf-f1c2-4d35-9f54-c41c09bd8265)
can you check on mention acc https://whitepaper-ft.meeds.io/portal/meeds/activity?id=254

**Comment by srenault-meeds** [comment]
Yes, so this is the same case I reported.
I will review the description and the expected outcome

**Comment by Jihed525** [comment]
+ reproduced also from user popover card 
![image](https://github.com/user-attachments/assets/61ee0b77-4c97-493c-94d2-4f3309c13970)

**Comment by boubaker** [comment]
Thanks @Jihed525 this will help to diagnose and fix it with a clear reproduction scenario.

**Comment by srenault-meeds** [comment]
@Jihed525 I have reviewed the description.

Now I do think it is not high priority as the recipient cannot access the message. WDYT?

**Comment by Jihed525** [comment]
@srenault-meeds how about greeting the effort of a collaborator in front of board of executives  ?
we might simply consider not notifying the receiver in this case, not valid ?

**Comment by srenault-meeds** [comment]
No, not valid. Imagine, someone is talking about you without you can know it. this is already the case we know that. But here we help doing it. This is agains my values

**Comment by Jihed525** [comment]
you're right, even if needed a simple post can do the job, agreed

**Comment by srenault-meeds** [comment]
Rejected as explained in previous comment
